weighted_graph/weight_graph.py

```diff
@@ -0,0 +1,94 @@
+# _*_ coding: utf-8 _*_
+"""Graph Data Structure."""
+
+
+class Graph(object):
+    """Class of the Graph Data Structure."""
+
+    def __init__(self):
+        """Initialize graph."""
+        self.graph = {}
+
+    def nodes(self):
+        """List nodes in graph."""
+        return list(self.graph.keys())
+
+    def edges(self):
+        """List edges in graph."""
+        lst = (list(self.graph.values()))
+        edge_list = []
+        for i in lst:
+            if i != []:
+                edge_list.extend(i)
+        return edge_list
+
+    def add_node(self, n):
+        """Add new node to the graph."""
+        self.graph[n] = {}
+        return self.graph
+
+    def add_edge(self, n1, n2, weight=0):
+        """Add edge between two nodes."""
+        if n1 in self.graph:
+            self.graph[n1].setdefault(n2, weight)
+        else:
+            self.graph.setdefault(n1, {n2: weight})
+            self.add_node(n2)
+
+    def del_node(self, n):
+        """Delete a node."""
+        if n in self.graph:
+            del self.graph[n]
+            return self.graph
```

Comment:
Shouldn't return anything. Also you have to make sure to delete the node from every other relationship in the graph